src/mesh/mesh.cpp

```diff
@@ -1149,83 +1132,64 @@ void Mesh::Initialize(int res_flag, ParameterInput *pin) {
     }
 
     if (res_flag == 0) {
-#pragma omp parallel for num_threads(nthreads)
-      for (int i = 0; i < nmb; ++i) {
-        MeshBlock *pmb = pmb_array[i];
-        pmb->ProblemGenerator(pin);
-      }
+      parallel_for("Problem Generator", block_range,
```

Comment:
@AndrewGaspar I see now - using `parallel_for` will launch `ProblemGenerator` inside a kernel. However, that won't work on the GPU, since `ProblemGenerator` will launch its own kernels and `*pin` is a host pointer, not a device pointer.

I think the Kokkos partition paradigm - which would give us CPU threading+GPU streams - is more appropriate here and is closer to the original intent.

Reply:
Maybe I accidentally obfuscated it, but these `parallel_for` calls target the `Kokkos::DefaultHostExecutionSpace`.

Reply:
I see, I missed that bit. That makes more sense